pilot/pkg/model/push_context.go

```diff
@@ -324,8 +324,33 @@ func (ps *PushContext) UpdateMetrics() {
 	}
 }
 
+// SetSidecarScope identifies the sidecar scope object associated with this
+// proxy and updates the proxy Node. This is a convenience hack so that
+// callers can simply call push.Services(node) while the implementation of
+// push.Services can return the set of services from the proxyNode's
+// sidecar scope or from the push context's set of global services. Similar
+// logic applies to push.VirtualServices and push.DestinationRule. The
+// short cut here is useful only for CDS and parts of RDS generation code.
+//
+// Listener generation code will still use the SidecarScope object directly
+// as it needs the set of services for each listener port.
+func (ps *PushContext) SetSidecarScope(proxy *Proxy) {
```

Comment:
Personally I prefer this to be of Proxy method 

func (*Proxy) SetSidecarScope (ps * PushContext) { 
}

Reply:
+1 - since it changes proxy.

Reply:
(but can be fixed in followup PR)